streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -121,21 +122,71 @@ public void shouldAvoidFlushingDeletionsWithoutDirtyKeys() {
     }
 
     @Test
-    public void shouldCloseAfterErrorWithFlush() {
+    public void shouldCloseWrappedStoreAfterErrorDuringCacheFlush() {
+        setUpCloseTests();
+        cache.flush(CACHE_NAMESPACE);
+        EasyMock.expectLastCall().andThrow(new NullPointerException("Simulating an error on flush"));
```

Comment:
Why not throw a general `RuntimeException` but a NPE?

Reply:
This was Bruno's test; I just left them alone.

Reply:
This was poorly stated, what I meant was that I was reluctant to change parts of the PR that had already been approved.